test/functional/NoOpContainerTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Dhii\Container\FuncTest;
+
+use Dhii\Container\NoOpContainer as Subject;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+class NoOpContainerTest extends TestCase
+{
+    /**
+     * @return MockObject&Subject
+     */
+    protected function createSubject()
+    {
+        $mock = $this->getMockBuilder(Subject::class)
```

Comment:
does it need to be a mock?

Reply:
Ya, SUT are usually mocked, because it helps with expectations.